### ISSUE TITLE
interface ISelect question

### ISSUE THREAD

**Issue by jyl087** [ujjwalguptaofficial/JsStore#23]
When I try to use a simple select, as shown in the example:

```
Connection.select({
   From: "Table_Name",
   OnSuccess:function (results){
       //results will be array of objects.
      console.log(results);
    },
   OnError:function (error) {
       alert(error.value);
    }
});
```
I get a "webpack: Failed to compile" error. It seems that the ISelect interface has all of its elements such as "From", "Where", "Skip", etc. as required. So, I am seeing "error TS2345" Argument of type '...' is not assignable to parameter of type 'ISelect'. Property 'Where' (or 'Skip' or 'Limit', etc) is missing in type ...'

So, it looks like I can only run "Connection.select" if I specify ALL parameters of the ISelect interface. I can't run the simple example as shown above. Is that right???

Thanks in advance.
Jim

**Comment by ujjwalguptaofficial** [comment]
yes you are right - all are required attribute which should not be. I am making others attribute optional. I will release this in next version.

Thanks

**Comment by udaychy** [comment]
For the time being - you can convert the Iselect to any like -

```
Connection.select({
   From: "Table_Name",
   OnSuccess:function (results){
       //results will be array of objects.
      console.log(results);
    },
   OnError:function (error) {
       alert(error.value);
    }
}as any);
```

It should work.

**Comment by jyl087** [comment]
ok, thanks. By the way, it's also strange that OnSuccess, OnError can be parameters in ISelect but also required as 2nd and 3rd arguments to the Connection.select method.

**Comment by ujjwalguptaofficial** [comment]
I think OnSuccess and OnError should not be orders, you can provide the options in any order.

Btw - if you don't like the OnSuccess and OnError - you can use promise. Check the doc - http://jsstore.net/tutorial/promise

**Comment by ujjwalguptaofficial** [comment]
I have solved the issue. Please install the new version - 1.6.0.

Let me know whats the results.

Thanks

**Comment by ujjwalguptaofficial** [comment]
Are you getting the issue anymore ? If not, please close the issue.

**Comment by jyl087** [comment]
Yes, the "as any" fixed the problem above. But, is that really a "fix"? If you want to close this issue, it's up to you. Thanks for your help, and great library!

**Comment by ujjwalguptaofficial** [comment]
We have released a new version - 1.6.0 . Are you getting the issue in this release too ?

**Comment by jyl087** [comment]
No, all good... thanks!
